evm/v0.5/testv2/Coordinator.test.ts

```diff
@@ -0,0 +1,909 @@
+import * as h from '../src/helpers'
+import { ethers } from 'ethers'
+import { LinkTokenFactory } from '../src/generated/LinkTokenFactory'
+import { CoordinatorFactory } from '../src/generated/CoordinatorFactory'
+import { EmptyAggregatorFactory } from '../src/generated/EmptyAggregatorFactory'
+import { MeanAggregatorFactory } from '../src/generated/MeanAggregatorFactory'
+import { GetterSetterFactory } from '../src/generated/GetterSetterFactory'
+import { MaliciousRequesterFactory } from '../src/generated/MaliciousRequesterFactory'
+import { MaliciousConsumerFactory } from '../src/generated/MaliciousConsumerFactory'
+import { Instance } from '../src/contract'
+import { makeTestProvider } from '../src/provider'
+import { assert } from 'chai'
+import { assertBigNum } from '../src/matchers'
+import { BigNumberish } from 'ethers/utils'
+
+const provider = makeTestProvider()
+
+const linkTokenFactory = new LinkTokenFactory()
+const coordinatorFactory = new CoordinatorFactory()
+const emptyAggregatorFactory = new EmptyAggregatorFactory()
+const meanAggregatorFactory = new MeanAggregatorFactory()
+const getterSetterFactory = new GetterSetterFactory()
+const maliciousRequesterFactory = new MaliciousRequesterFactory()
+const maliciousConsumerFactory = new MaliciousConsumerFactory()
+
+let roles: h.Roles
+
+const fs = (name: string) =>
+  h.getFunctionSignature(emptyAggregatorFactory, name)
+const partialServiceAgreement = {
+  aggInitiateJobSelector: fs('initiateJob'), // Currently, meanAggregator and emptyAggregator have the same signatures
+  aggFulfillSelector: fs('fulfill'),
+}
+
+async function initiateServiceAgreement(
+  coordinator: Instance<CoordinatorFactory>,
+  serviceAgreement: h.ServiceAgreement,
+) {
+  const signers = serviceAgreement.oracles.map(address =>
+    h.getWallet(address, roles),
+  )
+  const signatures = await h.generateOracleSignatures(serviceAgreement, signers)
+  return coordinator.initiateServiceAgreement(
+    h.encodeServiceAgreement(serviceAgreement),
+    h.encodeOracleSignatures(signatures),
+  )
+}
+
+const newServiceAgreement = async (
```

Comment:
I don't think this test helper is needed, I don't think the cost of additional indirection is worth the few lines saved. One change we could make to make it less verbose is to change the defaults of `newServiceAgreement` within `helpers.ts` to have the properties of `aggInitiateJobSelector` and `aggFulfillSelector` to their function signature hash values.